src/typed-geometry/functions/objects/intersection.hh

```diff
@@ -3154,5 +3206,249 @@ template <class ScalarT>
+    float dis_pos1 = dot(hs.normal, insec_sp.value().pos1) - distance(hs.center, pos<3, ScalarT>::zero); // signed distance plane extension - insec segment 2nd position
+
+    // cases depending on which segment point is inside of halfspace deduced by the hemisphere
+    if (dis_pos0 >= 0 && dis_pos1 >= 0)
+        return insec_sp;
+
+    if (dis_pos0 < 0 && dis_pos1 < 0)
+        return {};
+
+    auto insec_base_plane = intersection(s, base_plane);
+
+    if (!insec_base_plane.has_value())
+        return {};
+
+    if (dis_pos0 < 0) // replace first segment position
+        return segment<3, ScalarT>{insec_base_plane.value(), insec_sp.value().pos1};
+
+    if (dis_pos1 < 0) // replace second segment positon
+        return segment<3, ScalarT>{insec_sp.value().pos0, insec_base_plane.value()};
+
+    return {};
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr optional<segment<3, ScalarT>> intersection(hemisphere<3, ScalarT> const& hs, segment<3, ScalarT> const& s)
+{
+    return intersection(s, hs);
+}
+
+// sphere3 - plane3
+template <class ScalarT>
+[[nodiscard]] constexpr optional<disk<3, ScalarT>> intersection(sphere<3, ScalarT> const& s, plane<3, ScalarT> const& p)
+{
+    // get orthogonal vector for a given vector
+    auto const get_orthogonal = [](dir<3, ScalarT> normal) -> vec<3, ScalarT>
+    {
+        if (normal.x != ScalarT(0) || normal.y != ScalarT(0))
+            return vec<3, ScalarT>(cross(vec<3, ScalarT>(normal), vec<3, ScalarT>{0.f, 0.f, 1.f}));
+
+        else
+            return vec<3, ScalarT>(cross(vec<3, ScalarT>(normal), vec<3, ScalarT>{1.f, 0.f, 0.f}));
+    };
+
+    if (!intersects(s, p))
+        return {};
+
+    // project sphere center onto plane
+    tg::pos<3, ScalarT> disk_center = project(s.center, p);
+
+    // line in plane
+    auto dir = normalize(get_orthogonal(p.normal));
+    auto l = line<3, ScalarT>(pos<3, ScalarT>(p.normal * p.dis), dir);
+
+    auto s_bound = boundary_of(s);
+    // intersection with sphere boundary
+    auto insec = intersection(l, s_bound);
+
+    if (!insec.any())
+        return {};
+
+    // distance to disk_center = radius
+    float rad = distance(disk_center, insec.first());
+
+    return disk<3, ScalarT>(disk_center, rad, p.normal);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr optional<sphere<2, ScalarT, 3>> intersection(plane<3, ScalarT> const& p, sphere<3, ScalarT> const& s)
+{
+    return intersection(s, p);
+}
+
+// plane3 - inf_cylinder3
+template <class ScalarT>
+[[nodiscard]] constexpr optional<ellipse<2, ScalarT, 3>> intersection(plane<3, ScalarT> const& p, inf_cylinder<3, ScalarT> const& c)
+{
+    // early-out: plane normal and cylinder axis are orthogonal
+    if (dot(p.normal, c.axis.dir) == 0)
+        return {};
+
+    // ellipse mid_point = intersection cylinder axis and plane
+    auto insec_mid_axis = tg::intersection(c.axis, p);
+    if (!insec_mid_axis.any())
```

Comment:
Not semantically. The upper case should exclude all cases where the plane is parallel to the cylinder(-axis) because in these cases we won't get an ellipse as return.
2nd, I check if there is an intersection with the clinder-axis, which is also necessary to obtain an ellipse. Not all planes which are not parallel to the cylinder axis intersect with the cylinder.

Reply:
upper one is necessary to avoid that cylinder axis lies on plane, which would not result in an ellipse.

Reply:
But the only way the cylinder axis and the plane do not intersect is when plane normal and cylinder axis are orthogonal, or am I missing something here?

Reply:
true. The second check is already covered by the first. Missed that!